src/agent/base_agent.py

```diff
@@ -58,6 +59,7 @@ def __init__(
         user_persona_txt,
         past_chat_json,
         debug=True,
+        llm = None,
```

Comment:
What about choosing llm model with a string argument?
So, we can get the proper agent using this `agent_instance = getattr(sys.modules[__name__], agent_name)()`.